include/pros/optical.hpp

```diff
@@ -0,0 +1,232 @@
+	 * the operation failed, setting errno.
+	 */
+	std::int32_t get_led_pwm();
+
+	/**
+	 * Get the processed RGBC data from the sensor
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return rgb value if the operation was successful or PROS_ERR if
+	 * the operation failed, setting errno.
+	 */
+	pros::c::optical_rgb_s_t get_rgb();
+
+	/**
+	 * Get the raw un-processed RGBC data from the sensor
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return raw rgb value if the operation was successful or PROS_ERR if
+	 * the operation failed, setting errno.
+	 */
+	pros::c::optical_raw_s_t get_raw();
+
+	/**
+	 * Get the most recent gesture data from the sensor
+	 *
+	 * Gestures will be cleared after 500mS
+	 * 0 = no gesture
+	 * 1 = up (towards cable)
+	 * 2 = down
+	 * 3 = right
+	 * 4 = left
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return gesture value if the operation was successful or PROS_ERR if
+	 * the operation failed, setting errno.
+	 */
+	pros::c::optical_direction_e_t get_gesture();
+
+	/**
+	 * Get the most recent raw gesture data from the sensor
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return gesture value if the operation was successful or PROS_ERR if
+	 * the operation failed, setting errno.
+	 */
+	pros::c::optical_gesture_s_t get_gesture_raw();
+
+	/**
+	 * Enable gesture detection on the sensor
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return The Error code encountered
+	 */
+	std::int32_t enable_gesture();
+
+	/**
+	 * Disable gesture detection on the sensor
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of V5 ports (1-21).
+	 * ENODEV - The port cannot be configured as an Optical Sensor
+	 *
+	 * \return The Error code encountered
+	 */
+	std::int32_t disable_gesture();
+
+  /**
+	 * Gets the port number of the Optical Sensor.
+	 *
+	 * \return The Optical Sensor's port number.
+	 */
+	std::uint8_t get_port();
+
+  protected:
```

Comment:
We usually make this private, though I agree it really should be protected. We should probably go through and do a consistency check.

Reply:
So should this be private for this PR?